src/effects/EffectUI.cpp

```diff
@@ -1284,14 +1284,12 @@ std::shared_ptr<EffectInstance> EffectUIHost::InitializeInstance()
 
 void EffectUIHost::CleanupRealtime()
 {
-   bool noPriorState(mSubscription);
    mSubscription.Reset();
-   if (mSupportsRealtime && mInitialized) {
-      auto mpState = mwState.lock();
 
-      if (noPriorState && mpState) {
-         AudioIO::Get()->RemoveState(mProject, nullptr, mpState);
-         mpState.reset();
+   if (mSupportsRealtime && mInitialized) {
+      if (mpTempProjectState) {
+         AudioIO::Get()->RemoveState(mProject, nullptr, mpTempProjectState);
+         mpTempProjectState.reset();
```

Comment:
This makes the member variable `mwState` not really used so it could be removed from the class.

I think only the inner `if` condition is enough and `mInitialized` can be removed from the class.

Reply:
Let's do it separately sometime soon please

Reply:
ok